spock-core/src/main/java/org/spockframework/runtime/model/FeatureInfo.java

```diff
@@ -91,6 +94,54 @@ public void addBlock(BlockInfo block) {
     blocks.add(block);
   }
 
+  /**
+   * @since 2.4
+   */
+  @Beta
+  public List<IMethodInterceptor> getSetupInterceptors() {
```

Comment:
I thought about it, but decided against it, because I think that is more tricky as there the behavior is different. Do you just want to affect the methods of this spec or, also all the parent specs. On the spec level you are free to decide what you want to do, but I don't really want to provide variants for each case. 
What would you expect to happen? Do you think that it is a critical thing to have?

Reply:
The behavior should essentially be the same as attaching to the specification and checking whether the feature is the one I care about. So basically it should be effective for all such methods that are executed for the feature that the listener is attached to. So yes, I think I would also expect the methods of parent specs to be affected. Unless I overlook something.

> Do you think that it is a critical thing to have?

Well, I find it strange to have it for the ones added here, but not for the method interceptors and imagine they could be as useful as the ones already added here. "critical" not really, as you can always attach like now and check for the current feature.

Or maybe another idea. We could either give the method to the `add...` call on the feature or the feature to the `add...` call on the method, or both variants, so that the listener is scoped to both.

Reply:
> Or maybe another idea. We could either give the method to the `add...` call on the feature or the feature to the `add...` call on the method, or both variants, so that the listener is scoped to both.

Could you elaborate a bit more, I'm not sure I understand what you mean.

Reply:
`featureInfo.addSetupMethodInterceptor(theMethodInfo) { ... }`
and / or
`theMethodInfo.addInterceptor(theFeatureInfo) { ... }`

Reply:
Now, I'm even more confused.
`featureInfo.addSetupMethodInterceptor(methodInfo) { ... }` where does `methodInfo` come from and how would it be used internally?

🤔 we could just add untargeted `add*MethodInterceptor` that would apply to all methods of its kind and you'd have to decide in the interceptor if you want to intercept the particular method or not.

Reply:
I've added `public void addScopedMethodInterceptor(MethodInfo targetMethod, IMethodInterceptor interceptor)`